docs/src/howto-guides/child_device_firmware_management.md

```diff
@@ -0,0 +1,199 @@
+
+Handling the above mentioned responsibilities involve
+multiple interactions with thin-edge over its MQTT and HTTP APIs.
+In cases, where the child device connector is installed alongside thin-edge on the same device,
+these APIs can be accessed via a local IP or even `127.0.0.1`.
+The MQTT APIs are exposed via port 1883 and the HTTP APIs are exposed via port 8000.
+But, when the child device connector is running directly on the external child device,
+the MQTT and HTTP APIs of thin-edge need to be accessed over the network using its IP address,
+which is configured using the tedge config settings `mqtt.external.bind_address` or `mqtt.bind_address`.
+
+The `c8y-firmware-plugin` running on thin-edge, along with the `child-device-connector`
+running either on the thin-edge device itself or directly on the child device,
+provides firmware management support for that device.
+
+## Declare firmware management support of child device
+
+At first, the child device needs to declare that it supports firmware management from Cumulocity
+using the [supported operations API](../tutorials/supported_operations.md) of thin-edge
+by simply creating an empty operations file on the thin-edge device as follows:
+
+```shell
+touch /etc/tedge/operations/c8y/<child-device-id>/c8y_Firmware
+```
+
+This action will add `c8y_Firmware` as a new supported operation for the child device in Cumulocity.
+
+```admonish note
+This initial bootstrapping step needs to be performed on the thin-edge device directly
+and can not be done from the child device over the network.
+An over-the-network API for the same could be provided in future releases. 
+```
+
+# Handle firmware update requests from thin-edge
+
+Once the firmware management operation is enabled for child device,
+it device is ready to serve firmware update requests received from Cumulocity via thin-edge.
+
+Handling firmware update requests from thin-edge is a 5-step process:
+
+1. Subscribe to, and receive firmware update requests via MQTT
+1. Send an “executing” operation status update to acknowledge the receipt of the request via MQTT
+1. Download the firmware file update from the URL received in the request via HTTP
+1. Apply the firmware file update on the child device
+1. Send a “successful” operation status update via MQTT
+
+The following sections cover these steps in detail.
+
+### Subscribe to firmware update requests via MQTT
+
+The child device connector must subscribe to the `tedge/{CHILD_ID}/commands/req/firmware_update` MQTT topic
+to receive the firmware update requests from thin-edge.
+
+**Example:**
+
+```shell
+mosquitto_sub -h {TEDGE_DEVICE_IP} -t "tedge/{CHILD_ID}/commands/req/config_snapshot"
+```
+
+These requests arrive in the following JSON format:
+
+**Payload:**
+
+```json
+{
+    "id": "{OP_ID}",
+    "attempt": 1,
+    "name": "{FIRMWARE_NAME}",
+    "version": "{FIRMWARE_VERSION}",
+    "url":"http://{TEDGE_DEVICE_IP}:8000/tedge/file-transfer/tedge-child/firmware_update/{FILE_ID}",
```

Comment:
I think you can find a better name than my suggestion, but `8000` is indeed configurable.
```suggestion
    "url":"http://{TEDGE_DEVICE_IP}:{TEDGE_DEVICE_HTTP_PORT}/tedge/file-transfer/tedge-child/firmware_update/{FILE_ID}",
```